src/nilss.jl

```diff
@@ -323,11 +323,11 @@ function dudt_g_dgdu!(dudt, gsave, dgdu, nilssprob::NILSSProblem, y, p, iseg)
       end
     else
       if dg_val isa Tuple
-        dg[1](dg_val[1],u,uf.p,nothing,j)
-        copyto!(_dgdu, dg_val[1])
+        dg[1](dg_val[1],u,p,nothing,j)
+        copyto!(_dgdu, -dg_val[1])
```

Comment:
This is actually going to give an allocation. Can you do a short PR to make that `_dgdu .= -dg_val[1]`?